aquadoggo/src/graphql/queries/collection.rs

```diff
@@ -433,4 +774,425 @@ mod tests {
+            // @TODO: resolve issue https://github.com/p2panda/aquadoggo/issues/348
+            // This is where the test fails, the document returned from the store only contains
+            // the "lyrics" field and so when the resolver tries to get and unwrap that field
+            // there is a panic.
+            let data = query_songs(
+                &client,
+                song_schema.id(),
+                &format!("(first: 2, after: {end_cursor})"),
+                "",
+            )
+            .await;
+            assert_eq!(data["query"]["documents"].as_array().unwrap().len(), 1);
+            let end_cursor = data["query"]["endCursor"].clone();
+            println!("{end_cursor}");
+
+            println!("==FORTH QUERY==");
+
+            let data = query_songs(
+                &client,
+                song_schema.id(),
+                &format!("(first: 2, after: {end_cursor})"),
+                "",
+            )
+            .await;
+            assert_eq!(data["query"]["documents"].as_array().unwrap().len(), 0);
+        })
+    }
+
+    #[rstest]
+    fn paginated_query_over_collection_with_only_meta_fields_selected(key_pair: KeyPair) {
+        test_runner(|mut node: TestNode| async move {
+            // Publish some lyrics to the node.
+            let (lyric_schema, view_ids) = here_be_some_lyrics(&mut node, &key_pair).await;
+
+            // Publish some songs, which contain "title", "artist" and "lyrics" fields.
+            let (song_schema, _) =
+                here_be_some_karaoke_hits(&mut node, &view_ids, &lyric_schema, &key_pair).await;
+
+            // Init a GraphQL client we'll use to query the node.
+            let client = graphql_test_client(&node).await;
+
+            println!("==FIRST QUERY==");
+
+            let data = query_songs_meta_fields_only(&client, song_schema.id(), "(first: 4)").await;
+            assert_eq!(data["query"]["documents"].as_array().unwrap().len(), 3);
+
+            println!("==SECOND QUERY==");
+
+            let data = query_songs_meta_fields_only(&client, song_schema.id(), "(first: 2)").await;
+            assert_eq!(data["query"]["documents"].as_array().unwrap().len(), 2);
+            let end_cursor = data["query"]["endCursor"].clone();
+
+            println!("==THIRD QUERY==");
+
+            let data = query_songs_meta_fields_only(
+                &client,
+                song_schema.id(),
+                &format!("(first: 2, after: {end_cursor})"),
+            )
+            .await;
+            // @TODO: resolve issue https://github.com/p2panda/aquadoggo/issues/348
+            // This test fails at a different location from the above. There is no internal panic
+            // however an incorrect number of documents are returned. We are missing one.
+            assert_eq!(data["query"]["documents"].as_array().unwrap().len(), 1);
+            let end_cursor = data["query"]["endCursor"].clone();
+            println!("{end_cursor}");
```

Comment:
Query returns an unexpected number of documents when paginating over a collection of documents but only querying the meat fields.